### ISSUE TITLE
Imbalance in Images During Training

### ISSUE THREAD

**Issue by CVKim** [sjtuplayer/anomalydiffusion#8]
![image](https://github.com/sjtuplayer/anomalydiffusion/assets/90014998/4af98a58-366f-4792-a820-5d5673da307d)

Using the sample config hyperparameters as they are, and only the hazelnut dataset, the above phenomenon occurs.

What could be the reason?

**Comment by sjtuplayer** [comment]
To avoid the overfitting problem of the position encoder, it is encouraged to train all the anomaly types at the same time. And for this figure, columns 2 and 4 have good results. And I'm not sure which "phenomenon" you're referring to (column 1 or column 3)?

**Comment by CVKim** [comment]
Is it okay that columns 2, 3, and 4 are saved with a rotation applied to the images? Additionally, it seems that column 1 has been saved with a zoomed-in effect. Is there any issue with this, and what can be done to address it for columns 1 to 4?

**Comment by sjtuplayer** [comment]
During training, to avoid the overfitting of the spatial encoder, we apply data augmentation (random rotation, translation, and zoom-in) to the training data. Therefore, the saved images are rotated and zoomed-in. To save images individually, you can modify the saving operation in the function 'log_img' in main.py.

**Comment by CVKim** [comment]
Thank you for your help.

I respect you!

**Comment by boxbox2** [comment]
Hi, I only see the code for generating masks in the test_Unet.py , but I didn't find any code for generating anomaly images. Could you please tell me how you implemented it?
